src/torrent.cpp

```diff
@@ -247,6 +245,9 @@ bool is_downloading_state(int const st)
 		, m_outstanding_file_priority(false)
 		, m_complete_sent(false)
 	{
+		if (p.flags & torrent_flags::need_save_resume)
+			set_need_save_resume(torrent_handle::if_metadata_changed, true);
```

Comment:
I would explicitly do `m_need_save_resume_data = torrent_handle::if_metadata_changed` instead of introducing such a "bad" parameter as `in_constructor`.

Reply:
yeah, good point